src/VenueSearchIterator.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace FestivalsApi;
+
+use FestivalsApi\Result\VenueSearchResult;
+use GuzzleHttp\Exception\GuzzleException;
+use IteratorAggregate;
+use LogicException;
+use Traversable;
+use function get_class;
+use function is_array;
+
+class VenueSearchIterator implements IteratorAggregate
+{
+    protected VenueSearchResult $last_result;
+
+    protected ?int $page_size = null;
+
+    protected ?array $query = null;
+
+    protected int $request_count = 0;
+
+    public function __construct(protected FestivalsApiClient $client)
+    {
+    }
+
+    /**
+     * @throws FestivalsApiClientException if API client encounters an error
+     * @throws GuzzleException if Guzzle encounters an error
+     * @throws LogicException if no query set
+     */
+    public function getIterator(): Traversable
+    {
+        $this->throwIfNoQuerySet();
+
+        $this->query['size'] = $this->page_size;
+        $this->query['from'] = 0;
+        do {
+            $events = $this->makeApiCall();
+            foreach ($events as $event) {
+                yield $event;
+            }
+            $this->query['from'] += $this->page_size;
+        } while (count($this->last_result->getVenues()) === $this->page_size);
```

Comment:
Could avoid making it depend on the type of last_result here by having the `VenueSearchResult` / `EventSearchResult` implement `Countable`? Or we could just capture the count in `makeApiCall` - it looks like that's actually the only thing we use the `last_result` property for?